Infoblox 3.0.0

sschuur opened the pull request:
Next major update to the Infoblox Cloud solution.
   
   Change(s):
   - Added 11 playbooks for threat intelligence import, sending emails and enriching incidents. Added accompanying readme and images
   - Added 8 Analytic rules and removed 3 old ones 
   - Updated Solution_Infoblox.json and SolutionMetadata.json
   - Updated infoblox_logo.svg file

   Reason for Change(s):
   - New content
   
   Version Updated:
   - Removed all old analytic rules in case users want to keep the older versions easier as they work differently than new ones.

   Testing Completed:
   - In closed enviro

   Checked that the validations are passing and have addressed any issues that are present:
   - Am having trouble [running local KQL & YAML validations](https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally), I can resolve any that come up when checked here at least. .NET 7.0 and SDK 7.0.305 installed.
 
![image](https://github.com/Azure/Azure-Sentinel/assets/78623042/678ae390-5c9c-40ef-a658-1e2948342058)

   - Am also having trouble packaging. Following steps [here](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution). Packages the playbooks fine but no other content.

![image](https://github.com/Azure/Azure-Sentinel/assets/78623042/94d94db2-e379-4ae4-a1cf-28748768edb6)

Not sure how to fix the validation + packaging issues. Have ran the scripts in the past without issue. Sorry about that!

sschuur commented:
For DetectionsValidations - we changed the IDs from the old and new on purpose since the rules are quite different from each other even though they look for the same thing. Should they be the same ID anyway?
For KqlValidations - Analytics queries already look for latest Threat Intelligence data. Just a bit out of order from the sample queries given. Is this ok or needs to match?

Apologize for all the commits, could not get local validations to work.

v-vdixit commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/Infoblox-HighThreatLevelQueryNotBlockedDetected.yaml:
Hi @sschuur please revert the id change in this file as this file already existed and has only been updated, thanks!

v-vdixit commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/Infoblox-TI-SyslogMatchFound-URL.yaml:
```diff
@@ -0,0 +1,67 @@
+  let dt_lookBack = 1h;
+  let ioc_lookBack = 14d;
+  let TI = ThreatIntelligenceIndicator
+  | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
```
Hi @sschuur to fix the failing kql validation, can you please replace the line "_**| where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()**_" in query with below lines: -

  | where TimeGenerated >= ago(ioc_lookBack)
  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
  | where Active == true and ExpirationDateTime > now()
  
  also remoce **_| where Active == true_** from two lines below this to avoid repeated filtering. Below is a screenshot of changes for your reference, thanks.
  
  
![image](https://github.com/Azure/Azure-Sentinel/assets/115772287/caa87495-9853-44d8-8e48-05a0ff75c0e2)

v-vdixit commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/Infoblox-TI-CommonSecurityLogMatchFound-MalwareC2.yaml:
```diff
@@ -0,0 +1,68 @@
+  let ioc_lookBack = 14d;
+  let TI = ThreatIntelligenceIndicator
+  | where TimeGenerated >= ago(ioc_lookBack)
+  //Get most recent ingested indicator in case there are copies
```
Hi @sschuur can you please remove the comment from between these lines, the comment is being read as a separate line by the KQL validation tests and is resulting in failure.

v-vdixit commented:
Hi @sschuur thanks for the updates, we are working with team on reviewing analytic rules, will update you shortly, thanks!

sschuur commented:
Hi @v-vdixit, I can't give out API keys for our platform, so let me know if you need sample threat intelligence data returned by the API calls in the playbooks! Or any other data too. Wasn't sure how you guys deal with returned data/threat intelligence for playbooks.

v-vdixit commented:
Hi @sschuur we are working with the team for reviewing analytic rules and playbooks, will update shortly, thanks!

v-vdixit commented:
Hi @sschuur please add readme files with newly added playbooks, for reference you can check the following format - [Sample Playbook folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Dynatrace/Playbooks/Add_DynatraceApplicationSecurityAttackSourceIpThreatIntelligence)

sschuur commented:
hi @v-vdixit, the readme for the playbooks is in https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Solutions/Infoblox%20Cloud%20Data%20Connector/Playbooks/readme.md and describes all of them on one readme. Is this ok instead of having one for each playbook? They'd pretty much be copy-paste. 

There are screenshots for the Incident-Enrichment-Domains and Incident-Send-Email but the rest just import various threat intelligence datasets into Threat Intelligence. Should I add shots for that?


No prob, I updated the existing InfobloxCloudDataConnector-sampledata.csv file in https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Sample%20Data/CEF/InfobloxCloudDataConnector-sampledata.csv. Do you need that in json instead?

For testing the TI analytic rules I added [InfobloxSampleTIMatch.csv](https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Sample%20Data/Custom/InfobloxSampleTIMatch.csv) that matches the TI imported with the playbooks with the InfobloxCDC table but not sure it will help or not.. Let me know if you need some TI samples in a different format or whatnot.

Thank you!!

v-vdixit commented:
> hi @v-vdixit, the readme for the playbooks is in https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Solutions/Infoblox%20Cloud%20Data%20Connector/Playbooks/readme.md and describes all of them on one readme. Is this ok instead of having one for each playbook? They'd pretty much be copy-paste.
> 
> There are screenshots for the Incident-Enrichment-Domains and Incident-Send-Email but the rest just import various threat intelligence datasets into Threat Intelligence. Should I add shots for that?
> 
> No prob, I updated the existing InfobloxCloudDataConnector-sampledata.csv file in https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Sample%20Data/CEF/InfobloxCloudDataConnector-sampledata.csv. Do you need that in json instead?
> 
> For testing the TI analytic rules I added [InfobloxSampleTIMatch.csv](https://github.com/sschuur/Azure-Sentinel/blob/Infoblox-3.0.0/Sample%20Data/Custom/InfobloxSampleTIMatch.csv) that matches the TI imported with the playbooks with the InfobloxCDC table but not sure it will help or not.. Let me know if you need some TI samples in a different format or whatnot.
> 
> Thank you!!

Hi @sschuur thanks for updating and yes if all playbooks have the same functionality we can add one common readme file for them, we will check on the data you have added and check the working of analytic rules for it, we will provide you update by Friday, July 21st 2023, thanks!

v-vdixit commented:
Hi @sschuur we are ingesting data for analytic rules, will test the rules post that and provide an update by 24 Jul 2023

v-vdixit commented:
Hi @sschuur we are checking analytic rules changes, we will complete review for them by 26 Jul 2023, thanks!

v-prasadboke commented:
Hello @sschuur sorry for the inconvenience and delay in updates, But we are still investigating this PR and you can expect an update till 31 Jul 2023.

sschuur commented:
Hi @v-prasadboke and @v-vdixit, thank you for the updates! Is there anything I can do to help speed things up? We are on a bit of a deadline to get this out soon.

v-prasadboke commented:
Hello @sschuur, Thank you for letting us know. Will take this on and update you ASAP.

v-prasadboke commented:
Hello @sschuur, Sorry for the inconvenience and delay in PR review. This PR is still being investigated and will update you shortly before 03 Aug 2023. 
Thank you, 
Prasad.

v-prasadboke commented:
Hello @sschuur, I'm facing an issue after deploying the solution.
After deployment Playbook is breaking. The playbook blade doesnt showcases the Playbooks of infoblox solution.
Again I'm very sorry for the delay to get this PR merged. 

I'll update you soonest regarding this PR. 
Thanks, Prasad.

v-prasadboke commented:
Hello @sschuur, postdeployment and lastupdatetime is missing from all the playbooks, Kindly please update them all.
Thank you.

v-prasadboke commented:
Hello @sschuur, we are waiting for your updates. Thank you.

sschuur commented:
hi @v-prasadboke, I added lastupdatetime and postDeployment to each playbook. I apologize if that caused you issues! Thank you!

v-prasadboke commented:
Hello @sschuur, Thanks for committing the changes. Will review the content ang get back to you before 11 Aug 2023.

sschuur commented:
Hi @v-prasadboke, 

Yay! Great to hear. 

I added a quick release notes file, but it is not detailed. I will detail this much better going forward. We went thru several zip files over the last couple years for varying reasons and things got bunched together.

I successfully packaged with the V3 tool. 

Thank you!

v-prasadboke commented:
Hello @sschuur, we have made the necessary changes and approved the PR.
But still a few changes are remaining which will be done before this Solution gets published.
So for this before publishing this solution please drop a mail to v-prasadboke@microsoft.com and v-atulyadav@microsoft.com so that before this solution is pushed to live we could complete the remaining changes.

Thanks, 
Prasad Boke.

sschuur commented:
Hi @v-prasadboke,

Got it. I sent an email to you and @v-atulyadav.

Thank you so much!!
